go-controller/pkg/kubevirt/pod_test.go

```diff
@@ -0,0 +1,212 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+const vmName = "test-vm"
+
+var _ = Describe("Kubevirt Pod", func() {
+	const (
+		t0 = time.Duration(0)
+		t1 = time.Duration(1)
+		t2 = time.Duration(2)
+		t3 = time.Duration(3)
+		t4 = time.Duration(4)
+	)
+	runningKvSourcePod := runningKubevirtPod(t0)
+	postSuccessfulMigrationKvSourcePod := completedKubevirtPod(t1)
```

Comment:
>postSuccessfulMigrationKvSourcePod is a little confusing for me, if this would be the last pod the migration would be considered failed right?

That is indeed some convoluted wording, I apologize. What I meant to say is that pod was successfully migrated from, meaning it is now Completed. 
I'll reword it to `successfullyMigratedKvSourcePod`

>how is this pod treated differently from failedMigrationKvTargetPod?

well this function mainly looks at the target pod, so sourcePods (that can either be `Running` before/during migration and `Completed` after successful migration) doesn't really matter in terms of deciding the Migration State.